### PR TITLE
UCT/IFACE: Check RC verbs support by creating a dummy QP

### PR DESCRIPTION
**Pull request by SeyedMir**
## Why ?
The old way of checking support for RC verbs does not work correctly for EFA devices. It wrongly concludes that the device supports RC and thus leads to errors in `ucx_info -d`.

Fixes https://github.com/openucx/ucx/issues/6882

### CODE REVIEW

**Comment by shamisp** [comment]
ok to test

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 1]
```diff
@@ -494,16 +494,68 @@ static uct_rc_iface_ops_t uct_rc_verbs_iface_ops = {
     .ep_post_check = uct_rc_verbs_ep_post_check,
 };
 
+static ucs_status_t uct_rc_verbs_can_create_qp(struct ibv_context *ctx)
+{
+    struct ibv_pd *pd;
+    struct ibv_cq *cq;
+    struct ibv_qp *qp;
+    struct ibv_qp_init_attr qp_init_attr = {
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c:502]: initialized vars should be first

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 2]
```diff
@@ -494,16 +494,68 @@ static uct_rc_iface_ops_t uct_rc_verbs_iface_ops = {
...
 uct_rc_verbs_query_tl_devices(uct_md_h md,
                               uct_tl_device_resource_t **tl_devices_p,
                               unsigned *num_tl_devices_p)
 {
     uct_ib_md_t *ib_md = ucs_derived_of(md, uct_ib_md_t);
-    int flags;
+    ucs_status_t status;
+
+    /* device does not support RC if we cannot create an RC QP */
+    status = uct_rc_verbs_can_create_qp(ib_md->dev.ibv_context);
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c:553]: should pass pd from md instead of creating new one

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 3]
```diff
@@ -494,16 +494,68 @@ static uct_rc_iface_ops_t uct_rc_verbs_iface_ops = {
...
+        .cap.max_recv_sge    = 1,
+        .cap.max_inline_data = 0
+    };
+
+    ucs_status_t status = UCS_OK;
+
+    pd = ibv_alloc_pd(ctx);
+    if (!pd) {
+        ucs_error("failed to allocate pd %m");
+        return UCS_ERR_IO_ERROR;
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c:517]: pls add label "err:" and "goto err;" here to unify function's exit to one place

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 4]
```diff
@@ -494,16 +494,68 @@ static uct_rc_iface_ops_t uct_rc_verbs_iface_ops = {
...
+        .qp_type             = IBV_QPT_RC,
+        .sq_sig_all          = 0,
+        .cap.max_send_wr     = 1,
+        .cap.max_recv_wr     = 1,
+        .cap.max_send_sge    = 1,
+        .cap.max_recv_sge    = 1,
+        .cap.max_inline_data = 0
+    };
+
+    ucs_status_t status = UCS_OK;
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c:512]: why need to init?

File: src/uct/ib/rc/verbs/rc_verbs_iface.c [hunk 5]
```diff
@@ -494,16 +494,68 @@ static uct_rc_iface_ops_t uct_rc_verbs_iface_ops = {
...
+        status = UCS_ERR_IO_ERROR;
+        goto out_dealloc_pd;
+    }
+
+    qp_init_attr.send_cq = cq;
+    qp_init_attr.recv_cq = cq;
+
+    qp = ibv_create_qp(pd, &qp_init_attr);
+    if (!qp) {
+        status = UCS_ERR_UNSUPPORTED;
```
Comment (yosefe) [inline src/uct/ib/rc/verbs/rc_verbs_iface.c:532]: if (qp == NULL) {
  status = ...
  goto err_destroy_cq;
}

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**